docs/creating-test-environment.md

```diff
@@ -0,0 +1,194 @@
+# Creating the Clusters
+
+## Setup Control Plane and Runtime Equivalent
+
+### Local Setup
+
+```sh
+k3d cluster create op-skr --registry-create op-skr-registry.localhost
+k3d cluster create op-kcp --registry-create op-kcp-registry.localhost
+```
+
+### Use External Clusters
+
+Make sure to have two `KUBECONFIG` compliant client configurations at hand, one for kcp, one for skr
+
+## Setting up your registry
+
+### Make sure the registries are reachable via localhost
+
+Add the following to your `etc/hosts` entry.
+
+```/etc/hosts
+##
+# Host Database
+#
+# localhost is used to configure the loopback interface
+# when the system is booting.  Do not change this entry.
+##
+127.0.0.1       localhost
+255.255.255.255 broadcasthost
+::1             localhost
+
+# Added by Docker Desktop
+# To allow the same kube context to work on the host and the container:
+127.0.0.1 kubernetes.docker.internal
+
+# Added for Operator Registries
+127.0.0.1 op-kcp-registry.localhost
+127.0.0.1 op-skr-registry.localhost
+```
+
+### Using an external Registry
+
+TODO
+
+## Make sure you are in the Control Plane
+
+```
+kubectl config use k3d-op-kcp
+```
+
+## Install CRDs of Operator Stack
+
+_Note for Remote Clusters: Make sure you run the commands with `KUBECONFIG` set to the KCP Cluster_
+
+### Install Manifest Operator CRDs
+
+1. Checkout https://github.com/kyma-project/manifest-operator
+2. Run the Installation Command
+
+```
+make install
+```
+
+### Install Kyma Operator CRDs
+
+1. Checkout https://github.com/kyma-project/kyma-operator and `cd operator`
+2. Run the Installation Command
+
+```
+make install
+```
+
+Ensure the CRDs are installed with `k get crds | grep kyma-project.io`:
+
+```
+manifests.component.kyma-project.io        2022-08-18T16:27:21Z
+kymas.operator.kyma-project.io             2022-08-18T16:29:28Z
+moduletemplates.operator.kyma-project.io   2022-08-18T16:29:28Z
+```
+
+## Build your module
+
+In `https://github.com/kyma-project/kyma-operator`, `cd samples/template-operator`
+
+After this find the Port of your KCP OCI Registry and write it to `MODULE_REGISTRY_PORT`:
+
+### Using a Local Module/Image Registry
+
+```
+export MODULE_REGISTRY_PORT=$(docker port op-kcp-registry.localhost 5000/tcp | cut -d ":" -f2)
+export IMG_REGISTRY_PORT=$(docker port op-skr-registry.localhost 5000/tcp | cut -d ":" -f2)
+```
+
+### Using a Remote Module/Image Registry
+
+```
+export MODULE_REGISTRY=your-registry-goes-here.com
+export IMG_REGISTRY=your-registry-goes-here.com
+```
+
+### Generating and Pushing the Operator Image and Charts
+
+Next generate and push the module image of the operator
+
+```
+make module-operator-chart module-image
```

Comment:
This step doesn't work because there's a difference in host names.
In the `/etc/hosts` we have:
```
127.0.0.1 op-kcp-registry.localhost
```
While in the  Makefile the URLs are: `IMG_REGISTRY ?= op-kcp-registry:56888/operator-images`

I've added the missing `.localhost` suffix to the URL in the Makefile and the command worked OK.

Reply:
This is also not the case:

```
# Module Name used for bundling the OCI Image and later on for referencing in the Kyma Modules
MODULE_NAME ?= template
# Semantic Module Version used for identifying the build
MODULE_VERSION ?= 0.0.4
# Module Registry used for pushing the image
MODULE_REGISTRY_PORT ?= 50092
MODULE_REGISTRY ?= op-kcp-registry.localhost:$(MODULE_REGISTRY_PORT)/unsigned
# Desired Channel of the Generated Module Template
MODULE_TEMPLATE_CHANNEL ?= stable

# Image URL to use all building/pushing image targets
IMG_REGISTRY_PORT ?= $(MODULE_REGISTRY_PORT)
IMG_REGISTRY ?= op-skr-registry.localhost:$(IMG_REGISTRY_PORT)/unsigned/operator-images
IMG ?= $(IMG_REGISTRY)/$(MODULE_NAME)-operator:$(MODULE_VERSION)
```
shows the correct references with `.localhost` appended. Looking at previous comments maybe you dont have the latest state of the branch.